### ISSUE TITLE
Should there be format-top-level-form function?

### ISSUE THREAD

**Issue by sooheon** [cursive-ide/cursive#2236]
Related: #1080 

Not everyone reformats the entire file compulsively as they code. It's pretty bad form to make contributions where the indentation has changed in the entire file. I quite like the approach [ws-butler](https://github.com/lewang/ws-butler) or [lispy](https://www.github.com/abo-abo/lispy) take (see fn bound to lispy's `i`), where you can elect to reformat only the lines/forms you've touched.

I currently do this by selecting the form, and then reformatting, but it would be nice to have one shortcut to do this.

**Comment by cursive-ide** [comment]
Yes, I've been planning to add a "Reformat top-level form" action. 

IntelliJ actually provides some features for this: https://www.jetbrains.com/help/idea/reformat-file-dialog.html, in particular see "Only VCS changed text".

**Comment by sooheon** [comment]
Great, that's good enough for me for now.
